collect_app/src/main/java/org/odk/collect/android/formentry/backgroundlocation/FormEntryViewModel.java

```diff
@@ -0,0 +1,49 @@
+package org.odk.collect.android.formentry.backgroundlocation;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.ViewModel;
+
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.location.client.GoogleLocationClient;
+
+public class FormEntryViewModel extends ViewModel {
+    @NonNull
+    private final BackgroundLocationManager locationManager;
+
+    public FormEntryViewModel() {
+        locationManager = new BackgroundLocationManager(new GoogleLocationClient(Collect.getInstance().getApplicationContext()),
+                new BackgroundLocationHelper());
+    }
+
+    public void formFinishedLoading() {
```

Comment:
I feel like a nice next step here would be to have the ViewModel (or something it owns) coordinate form loading from disk. Probably a whole kettle of fish we want to throw out the kitchen right now (just go with me on that metaphor).

Reply:
yes, Yes, YES! And I think a state-machine-like approach would work very nicely for that as well, again keeping the ViewModel extremely thin and having another model object that keeps track of what work has been done to build up the form controller, what errors have come up, what needs to be displayed to the user, etc.